### PR TITLE
Kerberos update

### PR DESCRIPTION
**Pull request by gpotter2**
Important kerberos update:
- kerberos over TCP, reassembly
- add cryptographic utils from RFC3961, based on a modified version of a *previous* impacket's file, that itself a modified version of another implementation (i.e. BSD-2-Clause, compatible with GPL2)
- kerberos FAST (RFC6113)
- handle ancient RFC1964 APPLICATION0 tag
- handle KRB-ERROR with KDC_ERR_PREAUTH_REQUIRED
- add proper PADATA support
- tons of user-friendly improvements (enums...)

### CODE REVIEW

**Comment by gpotter2** [comment]
@guedou @p-l- Also ready for review !

File: scapy/libs/rfc3961.py [hunk 1]
```diff
@@ -0,0 +1,729 @@
...
+# Note that the following work is based only on THIS COMMIT from impacket,
+# which is therefore under mhorowitz's BSD 2-clause "simplified" license.
+
+import math
+import os
+import struct
+from scapy.compat import orb, chb, int_bytes, bytes_int, plain_str
+
+try:
+    from Cryptodome.Cipher import AES, DES3, ARC4, DES
```
Comment (gpotter2) [inline scapy/libs/rfc3961.py:23]: Cryptography doesn't support some of those (DES for instance), which is why we use cryptodome.. :/

**Comment by gpotter2** [comment]
Self merging (sorry) because I need this for a demo :/ and I'd like the doc to be built.
Feel free to review post-merge, I'll update it.

**Comment by p-l-** [comment]
@gpotter2 no worries; I did not have enough time to review it properly, sorry about that, but what I have read is great.

**Pull request merged**